src/psyclone/psyir/transformations/increase_rank_loop_arrays_trans.py

```diff
@@ -129,69 +130,101 @@ def validate(self, node, options=None):
         routine = node.ancestor(Routine)
         if routine is None:
             raise TransformationError(
-                f"The target of the {self.name} transformation should be a "
-                f"Loop inside a Routine.")
-
-        # Capture all symbols used inside codeblocks, this are not permited
+                f"The target Loop of the {self.name} transformation must be "
+                f"inside a Routine.")
+
+        # Check if the loop bound expressions are static
+        values_to_check = set()
+        for ref in (node.start_expr.walk(Reference) +
+                    node.stop_expr.walk(Reference)):
+            if ref.symbol.is_static:
+                continue
+            if isinstance(ref.parent, IntrinsicCall):
+                # The routine name of a intrinsic, or the arguments of inquiry
+                # functions don't need to be check
+                if ref.parent.is_inquiry or ref.position == 0:
+                    continue
+            values_to_check.add(ref.symbol)
+        for assignment in routine.walk(Assignment):
+            if isinstance(assignment.lhs, Reference):
+                if assignment.lhs.symbol in values_to_check:
+                    raise TransformationError(
+                        f"{self.name} can only be applied to loops with static"
+                        f" loop bound expressions, but in has been attempted "
+                        f"in a loop with the variable "
+                        f"'{assignment.lhs.symbol.name}'."
+                    )
+
+        # Capture all symbols used inside codeblocks, these are not permitted
         codeblock_symbols = set()
         for cb in routine.walk(CodeBlock):
             for name in cb.get_symbol_names():
                 codeblock_symbols.add(name.lower())
 
         # Each item listed in the array list must be a local Array Symbol or a
         # string that resolves to it
-        array_list = options.get("arrays", [])
-        if len(array_list) == 0:
+        if not isinstance(arrays, list) or len(arrays) == 0:
             raise TransformationError(
                 f"{self.name} has a mandatory 'arrays' option that "
```

Comment:
"...that is required to specify which arrays are to have their rank increased."

Reply:
Done